libs/storage/config.py

```diff
@@ -0,0 +1,201 @@
+                access_mode=DataVolume.AccessMode.RWO,
+                snapshot=True,
+                online_resize=True,
+                wffc=True,
+            ),
+            StorageClass(
+                name=StorageClassNames.NFS,
+                volume_mode=DataVolume.VolumeMode.FILE,
+                access_mode=DataVolume.AccessMode.RWX,
+                snapshot=False,
+                online_resize=False,
+                wffc=False,
+            ),
+            StorageClass(
+                name=StorageClassNames.OCI,
+                volume_mode=DataVolume.VolumeMode.BLOCK,
+                access_mode=DataVolume.AccessMode.RWX,
+                snapshot=True,
+                online_resize=True,
+                wffc=True,
+            ),
+            StorageClass(
+                name=StorageClassNames.OCI_UHP,
+                volume_mode=DataVolume.VolumeMode.BLOCK,
+                access_mode=DataVolume.AccessMode.RWX,
+                snapshot=True,
+                online_resize=True,
+                wffc=True,
+            ),
+            StorageClass(
+                name=StorageClassNames.RH_INTERNAL_NFS,
+                volume_mode=DataVolume.VolumeMode.FILE,
+                access_mode=DataVolume.AccessMode.RWX,
+                snapshot=True,
+                online_resize=True,
+                wffc=False,
+            ),
+        ]
+
+    def get_storage_config(self) -> dict[str, Any] | None:
+        for storage_class in self.supported_storage_classes():
+            if storage_class.name == self.name:
+                return asdict(obj=storage_class)
+
+        return None
+
+    def construct_storage_class_matrix(self, storage_config: str | None = None) -> list[dict[str, Any]]:
+        """
+        Constructs a storage class matrix from the storage class config
+
+        Args:
+            storage_config (str | None): Storage class config.
+                Format: `volume_mode=Block,access_mode=RWO,snapshot=True,online_resize=True,wffc=False`
+
+        Returns:
+            list[dict[str, str | bool]]: Storage class matrix
+
+        """
+        if sc_config := self.get_storage_config():
+            sc_config["default"] = True
+            return [{sc_config.pop("name"): sc_config}]
+
+        else:
+            LOGGER.info(f"Could not find storage class configuration for {self.name}. Constructing from user input")
+            cmd_config = {}
+
+            if storage_config:
+                if not all("=" in item for item in storage_config.split(",")):
+                    raise ValueError("Invalid format: all items must be key=value pairs, separated by comma")
+
+                cmd_config = dict(item.split("=") for item in storage_config.split(","))
+
+            sc_config = {
+                self.name: {
+                    "volume_mode": getattr(
+                        DataVolume.VolumeMode,
+                        cmd_config.get("volume_mode", "").title(),
```

Comment:
Should it be `upper()`? Because we want to get `DataVolume.VolumeMode.BLOCK` if `volume_mode=Block` was passed?